app/grandchallenge/datatables/views.py

```diff
@@ -54,7 +55,12 @@ def draw_response(self, *, form_data):
         order_by = f"{'-' if order_dir == 'desc' else ''}{order_by}"
         data = self.filter_queryset(self.object_list, search, order_by)
         paginator = self.get_paginator(queryset=data, per_page=page_size)
-        objects = paginator.page(page)
+
+        try:
+            objects = paginator.page(page)
+        except EmptyPage:
+            # If the page is out of range, show the last page
+            objects = paginator.page(paginator.num_pages)
```

Comment:
Not too sure if this is the solution. Is there a risk the page indicator could get out of sync?

Reply:
I don't think so: the page number doesn't exist anymore anyway in the page indicator ui: there simply aren't that many pages for that search result. Not sure how the JS handles the indicator internally, but I haven't been able to catch it going out of sync.

Reply:
Okay, doesn't feel great but lets see!